subsys/fb/cfb.c

```diff
@@ -116,16 +164,40 @@ int cfb_print(struct device *dev, char *str, u16_t x, u16_t y)
 		LOG_ERR("Wrong font size");
 		return -1;
 	}
-
-	if ((fb->screen_info & SCREEN_INFO_MONO_VTILED) && !(y % 8)) {
+	switch (fb->pixel_format) {
+	case PIXEL_FORMAT_MONO10:
+	case PIXEL_FORMAT_MONO01:
+		if ((fb->screen_info & SCREEN_INFO_MONO_VTILED)
+			&& !(y % 8)) {
+			for (size_t i = 0; i < strlen(str); i++) {
+				if (x + fptr->width > fb->x_res) {
+					x = 0U;
+					y += fptr->height;
+				}
+				x += fb->kerning +
+				draw_char_vtmono(fb, str[i], x, y);
+			}
+			return 0;
+		}
+		break;
+	case PIXEL_FORMAT_RGB_565:
 		for (size_t i = 0; i < strlen(str); i++) {
```

Comment:
Looks like a lot of code is duplicated here. Why not just select the correct draw_char function and keep the for loop common?

Reply:
right, will do that